dglgo/dglgo/model/graph_model/gin_ogbg.py

```diff
@@ -0,0 +1,106 @@
+import dgl
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from dgl.nn import GINEConv, AvgPooling, SumPooling
+from ogb.graphproppred.mol_encoder import AtomEncoder, BondEncoder
+
+class MLP(nn.Module):
+    def __init__(self,
+                 feat_size: int):
+        """Multilayer Perceptron (MLP)
+
+        Parameters
+        ----------
+        feat_size : int
+            Input and output feature size.
+        """
+        super(MLP, self).__init__()
+        self.mlp = nn.Sequential(
+            nn.Linear(feat_size, 2 * feat_size),
+            nn.BatchNorm1d(2 * feat_size),
+            nn.ReLU(),
+            nn.Linear(2 * feat_size, feat_size),
+            nn.BatchNorm1d(feat_size)
+        )
+
+    def forward(self, h):
+        return self.mlp(h)
+
+class OGBGGIN(nn.Module):
+    def __init__(self,
+                 data_info: dict,
+                 embed_size: int = 300,
+                 num_layers: int = 5,
+                 dropout: float = 0.5,
+                 virtual_node : bool = False):
+        """Graph Isomorphism Network (GIN) variant introduced in baselines
+        for OGB graph property prediction datasets
+
+        Parameters
+        ----------
+        data_info : dict
+            The information about the input dataset.
+        embed_size : int
+            Embedding size.
+        num_layers : int
+            Number of layers.
+        dropout : float
+            Dropout rate.
+        virtual_node : bool
+            Whether to use virtual node.
+        """
+        super(OGBGGIN, self).__init__()
+        self.data_info = data_info
+        self.embed_size = embed_size
+        self.num_layers = num_layers
+        self.virtual_node = virtual_node
+
+        self.node_encoder = AtomEncoder(embed_size)
+        self.edge_encoders = nn.ModuleList()
+        self.conv_layers = nn.ModuleList()
+
+        for _ in range(num_layers):
+            self.conv_layers.append(GINEConv(MLP(embed_size)))
+            self.edge_encoders.append(BondEncoder(embed_size))
+
+        self.dropout = nn.Dropout(dropout)
+        self.pool = AvgPooling()
+        self.pred = nn.Linear(embed_size, data_info['out_size'])
+
+        if virtual_node:
+            self.virtual_emb = nn.Embedding(1, embed_size)
+            nn.init.constant_(self.virtual_emb.weight.data, 0)
+            self.mlp_virtual = nn.ModuleList()
+            for _ in range(num_layers - 1):
+                self.mlp_virtual.append(MLP(embed_size))
+            self.virtual_pool = SumPooling()
+
+    def forward(self, graph, node_feat, edge_feat):
+        if self.virtual_node:
+            virtual_emb = self.virtual_emb(torch.zeros(graph.batch_size).long().to(graph.device))
```

Comment:
`virtual_emb = self.virtual_emb.expand(graph.batch_size, -1)` should do the job

Reply:
Replaced with `virtual_emb = self.virtual_emb.weight.expand(graph.batch_size, -1)`